cuda/solver/batch_bicgstab_kernels.cu

```diff
@@ -33,21 +33,40 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "core/solver/batch_bicgstab_kernels.hpp"
 
 
+#include <thrust/functional.h>
+#include <thrust/transform.h>
```

Comment:
Yes, mainly due to included files from batch_multi_vector_kernels

Reply:
I think it can be hard to figure out which kernel file needs which includes. Perhaps it makes sense if there is the kernel implementation in `some_kernels.cpp.inc` then to have a file `some_kernels_includes.hpp.inc` (or just use `.hpp.inc`). But that might be done later.

Reply:
maybe add that to the to-do list